src/compiler/scala/tools/nsc/settings/ScalaSettings.scala

```diff
@@ -254,7 +254,7 @@ trait ScalaSettings extends AbsScalaSettings
 
     private val inlineChoices = List(lMethod, inline)
     val lInline = Choice("l:inline",
-      "Enable cross-method optimizations: " + inlineChoices.mkString("", ",", "."),
+      "Enable cross-method optimizations (note: inlining requires -opt-inlnie-from): " + inlineChoices.mkString("", ",", "."),
```

Comment:
I agree, and that's the current implementation. We could bring back the functionality of `-opt:l:project` with a special pattern in `-opt-inline-from:...:<sources>:...`